src/context/reducer.js

```diff
@@ -0,0 +1,73 @@
+import { createContext, useEffect, useReducer } from "react";
+
+import { firebaseAuth, firebaseDb } from "@/firebase/config";
+import { onAuthStateChanged } from "firebase/auth";
+import { doc, getDoc } from "firebase/firestore";
+
+export const InitialState = {
+  user: {
+    uid: "",
+    email: "",
+    displayName: "",
+  },
+  authState: false,
+};
+
+export const authReducer = ( state, action) => {
+  if (action.type === "SIGN_IN") {
+    return {
+      ...state,
+      user: action.payload,
+    };
+  }
+  if (action.type === "LOG_IN") {
```

Comment:
'SIGN_IN' and 'LOG_IN' are doing same thing

Reply:
@ibimina please explain why we have both sign_in and log_in doing the same thing 🙏🏾

Reply:
I have update the function